packages/editor-ui/src/views/NodeView.vue

```diff
@@ -2620,7 +2620,7 @@ export default mixins(
 					return;
 				}
 
-				if (trackHistory) {
+				if (trackHistory && !partOfBulkDelete) {
```

Comment:
why do we need this? should not it just be bundled as part of current bulk action?

Reply:
Nope, bulk action inside this function is used when deleting a single connected node. In that case, it will contain `removeNode` and `removeConnection` commands. When deleting multiple nodes, this function is called in a loop and recording is started outside of this loop, so starting and stopping recording in each iteration will break the history recording.

Reply:
gotcha